code/modules/research/secret_techs_vr.dm

```diff
@@ -0,0 +1,73 @@
+/*
+	The secret tech system! This is to add some more gameplay to RnD beyond the basic tech speedruns and deconstructing.
+
+	Basically, we setup origin techs on an item and also secret techs, when the item is successfully scanned in a radiocarbon spectrometer (code/modules/xenoarcheaology/tools/geosample_scanner.dm), then it will add secret techs to it's origin tech list.
+*/
+
+// Here we set the tech cap, this is the maximum tech that an item can have in it's default origin techs. Any higher and the difference is placed into secret tech.
+#define TECH_CAP 4
+
+/obj/item
+	var/list/secret_tech = null // This is a list of the techs that will be descovered in the spectrometer.
+	var/cap_override = FALSE // Set an item's cap override to true if you want it to not use the tech cap system, good for POI rewards and archeology finds! Note: Giving an item a secret tech list will do the same thing.
+
+// This is the proc for updating tech, calling it will add secret_tech into origin_tech, then set secret_tech to null so that one can't just infinitely increase an item's tech level.
+/obj/item/proc/update_techs()
+	if(!secret_tech) // Sanity check
+		return
+
+	if(!origin_tech)
+		origin_tech = list()
+	for(var/T in secret_tech)
+		if(T in origin_tech)
+			origin_tech[T] = secret_tech[T] + origin_tech[T]
+		else
+			origin_tech.Add(T)
+			origin_tech[T] = secret_tech[T] + origin_tech[T]
+	secret_tech = null
+
+// This gives us a hardcap at initialization.
+/obj/item/Initialize()
+	..()
```

Comment:
initialize absolutely needs to return an INITIALIZE_HINT

use
. = ..() for the default